libs/remix-ai-core/src/agents/contractAgent.ts

```
@@ -0,0 +1,142 @@
+import { parse } from "path";
+import { AssistantParams } from "../types/models";
+
+const compilationParams = {
+  optimize: false,
+  evmVersion: null,
+  language: 'Solidity',
+  version: '0.8.28+commit.7893614a'
+}
+
+interface CompilationResult {
+  compilationSucceeded: boolean
+  errors: string
+}
+
+export class ContractAgent {
+  plugin: any;
+  readonly generationAttempts: number = 3
+  nAttempts: number = 0
+  generationThreadID: string= ''
+  workspaceName: string = ''
+  contracts: any = {}
+
+  constructor(props) {
+    this.plugin = props;
+    AssistantParams.provider = this.plugin.assistantProvider
+  }
+
+  async writeContracts(payload, userPrompt) {
+    const currentWorkspace = await this.plugin.call('filePanel', 'getCurrentWorkspace')
+    try {
+      console.log('Writing contracts', payload)
+      this.contracts = {}
+      const parsedFiles = payload
+      this.generationThreadID = parsedFiles['threadID']
+      this.workspaceName = parsedFiles['projectName']
+
+      this.nAttempts += 1
+      console.log('Attempts', this.nAttempts)
+      console.log('Generation attempts', this.generationAttempts)
+      if (this.nAttempts > this.generationAttempts) {
+        console.error('Failed to generate the code')
+        return "Failed to generate secure code on this prompt ```" + userPrompt + "````"
```

this should throw an error instead of returning a string.